code/datums/spells/lichdom.dm

```diff
@@ -0,0 +1,129 @@
+/obj/effect/proc_holder/spell/targeted/lichdom
+	name = "Bind Soul"
+	desc = "A dark necromantic pact that can forever bind your soul to an item of your choosing. So long as both your body and the item remain intact and on the same plane you can revive from death, though the time between reincarnations grows steadily with use."
+	school = "necromancy"
+	charge_max = 10
+	clothes_req = 0
+	centcom_cancast = 0
+	invocation = "NECREM IMORTIUM!"
+	invocation_type = "shout"
+	range = -1
+	level_max = 0 //cannot be improved
+	cooldown_min = 10
+	include_user = 1
+
+	var/obj/marked_item
+	var/mob/living/current_body
+	var/resurrections = 0
+	var/existence_stops_round_end = 0
+
+	action_icon_state = "skeleton"
+
+/obj/effect/proc_holder/spell/targeted/lichdom/New()
+	if(!config.continous_rounds)
+		existence_stops_round_end = 1
+		config.continous_rounds = 1
```

Comment:
TG has something similar to what you describe, but I don't have the time, energy, or want-to to refactor gamemode code and port over Mulligan, so this is the nearest equal that doesn't require a lot of changes to function.